airflow/providers/weaviate/hooks/weaviate.py

```diff
@@ -133,20 +140,201 @@ def create_class(self, class_json: dict[str, Any]) -> None:
+        class_names = [class_names] if class_names and isinstance(class_names, str) else class_names
+
+        failed_class_list = []
+        for class_name in class_names:
+            try:
+                for attempt in Retrying(
+                    stop=stop_after_attempt(3),
+                    retry=retry_if_exception(lambda exc: self.check_http_error_is_retryable(exc)),
+                ):
+                    with attempt:
+                        client.schema.delete_class(class_name)
+            except Exception as e:
+                if if_error == "continue":
+                    self.log.error(e)
+                    failed_class_list.append(class_name)
+                elif if_error == "stop":
+                    raise e
+
+        if if_error == "continue":
+            return failed_class_list
+        return None
+
+    def delete_all_schema(self):
+        """Remove the entire schema from the Weaviate instance and all data associated with it."""
+        client = self.get_client()
+        return client.schema.delete_all()
+
+    def update_config(self, class_name: str, config: dict):
+        """Update a schema configuration for a specific class."""
+        client = self.get_client()
+        client.schema.update_config(class_name=class_name, config=config)
+
+    def upsert_classes(self, schema_json: dict[str, Any] | str, existing: ExitingSchemaOptions = "ignore"):
+        """
+        Create or update the classes in schema of Weaviate database.
+
+        :param schema_json: Json containing the schema. Format {"class_name": "class_dict"}
+            .. seealso:: `example of class_dict <https://weaviate-python-client.readthedocs.io/en/v3.25.2/weaviate.schema.html#weaviate.schema.Schema.create>`_.
+        :param existing: Options to handle the case when the classes exist, possible options
+            'replace', 'fail', 'ignore'.
+        """
+        existing_schema_options = ["replace", "fail", "ignore"]
+        if existing not in existing_schema_options:
+            raise ValueError(f"Param 'existing' should be one of the {existing_schema_options} values.")
+        if isinstance(schema_json, str):
+            schema_json = cast(dict, json.load(open(schema_json)))
+        set__exiting_classes = {class_object["class"] for class_object in self.get_schema()["classes"]}
+        set__to_be_added_classes = {key for key, _ in schema_json.items()}
+        intersection_classes = set__exiting_classes.intersection(set__to_be_added_classes)
+        classes_to_create = set()
+        if existing == "fail" and intersection_classes:
+            raise ValueError(
+                f"Trying to create class {intersection_classes}" f" but this class already exists."
+            )
+        elif existing == "ignore":
+            classes_to_create = set__to_be_added_classes - set__exiting_classes
+        elif existing == "replace":
+            error_list = self.delete_classes(class_names=list(intersection_classes))
```

Comment:
Yes, this operation is not atomic since the operation takes place in different transactions. But the whole operation is idempotent, so a user can retry the task.

But this is a very common operation when building LLM workflows as we have observed and will be useful to users while developing the LLM pipeline. But to make the risk explicit we can maybe add a proper disclaimer in the docstring, regarding the same. WDYT?

Also, I'll change the name of the method to `create_or_replace_classes`

Reply:
Yeah, the disclaimer sounds good to me

Reply:
I have renamed the methods, as per our discussion, that should also be sufficient.